packages/flutter_tools/lib/src/commands/build_windows.dart

```diff
@@ -19,11 +20,22 @@ import 'build.dart';
 /// A command to build a windows desktop target through a build shell script.
 class BuildWindowsCommand extends BuildSubCommand {
   BuildWindowsCommand({
+    required OperatingSystemUtils operatingSystemUtils,
     bool verboseHelp = false,
-  }) : super(verboseHelp: verboseHelp) {
+  }) : _operatingSystemUtils = operatingSystemUtils,
+       super(verboseHelp: verboseHelp) {
     addCommonDesktopBuildOptions(verboseHelp: verboseHelp);
+    // As long as arm64 does not have artifacts available, stick to x64.
+    final String defaultTargetPlatform = 'windows-x64';
+    argParser.addOption('target-platform',
+      defaultsTo: defaultTargetPlatform,
+      allowed: <String>['windows-arm64', 'windows-x64'],
+      help: 'The target platform for which the app is compiled.',
+    );
```

Comment:
Please add unit tests to `/packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart` that verify this option works as expected